website/views.py

```diff
@@ -23,4 +29,129 @@ def oh(request):
     return render(request, 'website/oh.html', {})
 
 def ir(request):
-	return render(request, 'website/ir.html', {})
\ No newline at end of file
+    return render(request, 'website/ir.html', {})
+
+def interview(request):
+    time_dict = {9: "9:00am - 10:00am", 10: "10:00am - 11:00am", 11: "11:00am - 12:00pm", 12: "12:00pm - 1:00pm", 13: "1:00pm - 2:00pm",
+                14: "2:00pm - 3:00pm", 15: "3:00pm - 4:00pm", 16: "4:00pm - 5:00pm"}
+    days_of_week = [0, 1, 2, 3, 4, 5, 6]
+    start_times = {0: 9, 1: 10, 2: 11, 3: 12, 4: 13, 5: 14, 6: 15, 7: 16}
+    interview_slot_list = InterviewSlot.objects.all()
+
+    time_slot_dict = {}
+    start_time = 9
+    for _ in range(len(time_dict)):
+        filter_start_time = interview_slot_list.filter(hour=start_time)
+        imputed_start_time = []
+        for day in days_of_week:
+            slot = filter_start_time.filter(day_of_week=day)
+            if len(slot) == 0:
+                imputed_start_time.append(None)
+            else:
+                #There should only be one slot from this filter 
+                imputed_start_time.append(slot[0])
+        time_slot_dict[start_time] = imputed_start_time
+        start_time += 1
+
+    now = datetime.now()
+    date = datetime.day
+
+    current_week = get_dates_of_week(now)
+    print(current_week)
+
+    context = {'interview_slot_list': interview_slot_list, 'day': now.strftime("%A"), 'date': date, 
+    "time_dict": time_dict, "time_slot_dict": time_slot_dict, "start_times": start_times, 
+    "range": range(len(start_times)), "week": current_week}
+
+    return render(request, 'website/interview.html', context)
+
+def get_dates_of_week(now):
```

Comment:
for these helper methods, better to have them start with an underscore like _get_dates_of_week so people don't try to call them in urls.py for some reason